src/universal/components/SignUpPage/SignUpPage.js

```diff
@@ -0,0 +1,119 @@
+/**
+ * The sign-up page.  Can sign up via email/password or 3rd party auth.
+ *
+ * @flow
+ */
+
+import type {Credentials} from 'universal/types/auth';
+import type {Dispatch} from 'redux';
+
+import React, {Component} from 'react';
+import {connect} from 'react-redux';
+
+import AuthPage from 'universal/components/AuthPage/AuthPage';
+import loginWithToken from 'universal/decorators/loginWithToken/loginWithToken';
+import withAtmosphere from 'universal/decorators/withAtmosphere/withAtmosphere';
+import {THIRD_PARTY_AUTH_PROVIDERS} from 'universal/utils/constants';
+import getWebAuth from 'universal/utils/getWebAuth';
+import SignUp from './SignUp';
+
+type Props = {
+  atmosphere: Object,
+  dispatch: Dispatch<*>,
+  hasSession: boolean
+};
+
+type State = {
+  error: ?{description: string},
+  submittingCredentials: boolean
+};
+
+class SignUpPage extends Component<Props, State> {
+  state = {
+    error: null,
+    submittingCredentials: false
+  };
+
+  getHandlerForThirdPartyAuth = (auth0Connection: string) => () => {
+    this.webAuth.authorize({
+      connection: auth0Connection,
+      responseType: 'token'
+    });
+  };
+
+  auth0SignUp = ({email, password}: Credentials): Promise<void> => {
+    return new Promise((resolve, reject) => {
+      this.webAuth.signup({
+        email,
+        password,
+        connection: 'Username-Password-Authentication',
+        responseType: 'token'
+      }, (error) => {
+        if (error) {
+          reject(error);
+        } else {
+          resolve();
+        }
+      });
+    });
+  };
+
+  auth0LogIn = (credentials: Credentials): Promise<void> => {
+    return new Promise((resolve, reject) => {
+      this.webAuth.login({
+        ...credentials,
+        realm: 'Username-Password-Authentication',
+        responseType: 'token'
+      }, (error) => {
+        if (error) {
+          reject(error);
+        } else {
+          resolve();
+        }
+      });
+    });
+  };
+
+  webAuth = getWebAuth();
+
+  handleSubmitCredentials = async (credentials: Credentials): Promise<void> => {
+    try {
+      this.setState({submittingCredentials: true, error: null});
+      await this.auth0SignUp(credentials);
+      await this.auth0LogIn(credentials);
+    } catch (error) {
+      this.setState({error});
```

Comment:
I'm not sure if this will ever be called on an unmounted component. This callback gets bound to the form submit event on a rendered child. The child's rendered tree can't be on the page in the first place if this component is unmounted.